### ISSUE TITLE
`control_statement` kills inline comments

### ISSUE THREAD

**Issue by herrernst** [realm/SwiftLint#6207]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

`control_statement` kills inline comments

Code before:

```c
if (abc == 1/* && cdf == 2*/) {
    print("Hello, world!")
}
```

Run SwiftLint
```bash
swiftlint --fix
```

Code afterwards:
```c
if abc == 1 {
    print("Hello, world!")
}
```

I would prefer it to keep comments and thus be:
```c
if abc == 1/* && cdf == 2*/ {
    print("Hello, world!")
}
```

See repo here: https://github.com/herrernst/swiftlint-control_statement-comments

### Environment

* SwiftLint version (run `swiftlint version` to be sure): 0.59.1
* Xcode version (run `xcodebuild -version` to be sure): Xcode 16.4 (Build version 16F6)
* Installation method used (Homebrew, CocoaPods, building from source, etc): SPM
* Configuration file: none (default configuration)

**Comment by theamodhshetty** [comment]
Opened #6545 for this.

I took the conservative route here: instead of trying to reposition inline comment trivia while removing the parentheses, the rule now skips `control_statement` corrections when the parenthesized condition contains comments. That keeps `swiftlint --fix` from deleting comments in cases like the one from this report.
